.github/workflows/c-cpp.yml

```diff
@@ -0,0 +1,36 @@
+name: C/C++ CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  actions: read
+  checks: read
+  contents: read
+  issues: write
+  pull-requests: write
+
+jobs:
+  build:
+
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: install dependencies
+      run: sudo apt-get install libcurl4-openssl-dev libboost-dev libboost-system-dev libboost-thread-dev
+    - name: init submodule
+      run: git submodule update --init --recursive
+    - name: download prebuilt
+      run: QuoteGeneration/download_prebuilt.sh
+    - name: download SGXSDK
+      run: |
+        curl https://download.01.org/intel-sgx/sgx-linux/2.24/distro/ubuntu22.04-server/sgx_linux_x64_sdk_2.24.100.3.bin -o ./sgx_linux_x64_sdk.bin
```

Comment:
Can we always use the latest SDK (with the link of latest in 01.org)?

Reply:
Latest version also has a version number in the installer file name. We also have to update for each release.

Reply:
Does curl support some wildcard (like * or ?) to ignore the specific version?

Reply:
> Does curl support some wildcard (like * or ?) to ignore the specific version?

I think it doesn't support. But maybe we can pass the version to this script from the environment?

Reply:
> I think it doesn't support. But maybe we can pass the version to this script from the environment?
I don't find any full version string of Linux SGXSDK in this repo. Only a Windows one in `DCAP_Components.nuspec`.

Reply:
Maybe we can add a link to the latest SDK installer in 01.org

Reply:
Can we use "wget"? It can support wildcard like:
wget -r -l1 -np -nd --accept 'sgx_linux_x64_sdk_*.bin' https://download.01.org/intel-sgx/latest/linux-latest/distro/ubuntu22.04-server/

Reply:
Perhaps we can update the version.xml on 01.org to add the full version. Then we can retrieve the version and construct the right curl command to avoid changing this action file for each release.

Reply:
> Can we use "wget"? It can support wildcard like: wget -r -l1 -np -nd --accept 'sgx_linux_x64_sdk_*.bin' https://download.01.org/intel-sgx/latest/linux-latest/distro/ubuntu22.04-server/

Looks like a good solution. I'll update.